src/graph/node_initializers.h

```diff
@@ -35,140 +37,229 @@ class NodeInitializer {
+ */
 Ptr<NodeInitializer> glorotNormal(bool fanIn = false, bool fanOut = false, float scale = 1.f);
 
-// @TODO: add documentation
-Ptr<NodeInitializer> dropout(float dropoutProbabilty);
+/**
+ * @brief Initialize a dropout mask (a tensor of 0 and 1) with given dropout probability.
+ * <a href=https://www.cs.toronto.edu/~hinton/absps/JMLRdropout.pdf>Dropout</a>
+ * is proposed as a technique to prevent Neural Networks from overfitting.
+ * @param dropoutProbability a float type defines the dropout probability.
+ *        E.g., dropoutProbability=0.1 means 90% of values are kept.
+ * @return A NodeInitializer
+ */
+Ptr<NodeInitializer> dropout(float dropoutProbability);
 
 /**
- * Intialize with gumbel noise, i.e. -log(-log(u)) where u ~ Uniform(0 + eps, 1 - eps)
- *
+ * Initialize with gumbel noise, i.e. -log(-log(u)) where u ~ Uniform(0 + eps, 1 - eps).
  * @return A NodeInitializer
  */
 Ptr<NodeInitializer> gumbel(float eps = 1e-5f);
 
-// @TODO: add documentation
+/**
+ * Initialize tensor by *copying* from the given vector.
+ * Creates a NodeInitializer that will initialize the tensor
+ * by *copying* the values from the given vector
+ * @return A NodeInitializer
+ */
 template <typename T>
 Ptr<NodeInitializer> fromVector(const std::vector<T>& v);
+
+/**
+ * Initialize tensor by *moving* from the given vector.
+ * Creates a NodeInitializer that will initialize the tensor
+ * by *moving* the values from the given vector into this tensor,
+ * and the given vector may be emptied.
+ * This version is the
+ * <a href=https://en.cppreference.com/w/cpp/language/reference>
+ * rvalue reference</a> overloading.
+ * @return A NodeInitializer
+ */
 template <typename T>
 Ptr<NodeInitializer> fromVector(std::vector<T>&& v);
 
-// @TODO: add documentation
+/**
+ * Initialize tensor from a given sparse vector.
+ * Creates a NodeInitializer that will initialize the tensor
+ * from a given sparse vector (stored in std::pair). The resulting
+ * tensor is first filled with `1e-6` (a placeholder for non-zero
+ * element), then set the value to the given sparse vector.
+ * @param v the sparse vector is stored in std::pair where
+ * the first object (v.first) holds the indexes (in a vector),
+ * and the second object (v.second) holds the corresponding
+ * values (in a vector).
+ * This means the value of the resulting tensor at index v.first[i]
+ * is v.second[i].
+ * @return A NodeInitializer
+ */
 Ptr<NodeInitializer> fromSparseVector(std::pair<std::vector<size_t>, std::vector<float>>& v);
 
-// @TODO: add documentation
+/**
+ * Initialize tensor by copying from the given io::Item.
+ * Creates a NodeInitializer that will initialize the tensor
+ * by copying the values from the given io::Item.
+ * If this io::Item is a memory-mapped item, then the function
+ * will set the memory pointing to this item.
+ * If this io::Item is a regular item, then the function will
+ * copy the values from this item.
+ * @return A NodeInitializer
+ */
```

Comment:
As above, check if formatting can be improved and if we can save vertical space.